pkg/consensus/retry_watcher.go

```diff
@@ -30,16 +34,29 @@ type retryWatcher struct {
 	getOpts   *etcd.GetOptions
 	watchOpts *etcd.WatcherOptions
 
+	refreshPeriod time.Duration
+	refreshNext   time.Time
+	now           func() time.Time
+
 	cur etcd.Watcher
 }
 
 func (w *retryWatcher) Next(ctx context.Context) (*etcd.Response, error) {
+	// Periodically force a full tree refresh.
+	if w.now().After(w.refreshNext) &&
```

Comment:
nit: unnecessary newline, and the `refreshNext.IsZero` & `refreshPeriod != 0` checks don't appear needed, right? isn't it enough to be after `refreshNext`?

With a channel strategy, you'd do a non-blocking select here and iff it selects, nil `w.cur`.